src/contract_wrappers/tokens/__tests__/security_token_wrapper.test.ts

```diff
@@ -0,0 +1,2862 @@
+
+      // Real call
+      const result = await target.balanceOfAt(mockedParams);
+      // Result expectation
+      expect(result).toBe(expectedResult);
+      // Verifications
+      verify(mockedContract.balanceOfAt).once();
+      verify(
+        mockedMethod.callAsync(mockedParams.investor, objectContaining(numberToBigNumber(mockedParams.checkpointId))),
+      ).once();
+    });
+  });
+
+  describe('getVersion', () => {
+    test('should call to getVersion', async () => {
+      const expectedResult = new BigNumber(1);
+      // Mocked method
+      const mockedMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.getVersion).thenReturn(instance(mockedMethod));
+      // Stub the request
+      when(mockedMethod.callAsync()).thenResolve(expectedResult);
+
+      // Real call
+      const result = await target.getVersion();
+      // Result expectation
+      expect(result).toBe(expectedResult);
+      // Verifications
+      verify(mockedContract.getVersion).once();
+      verify(mockedMethod.callAsync()).once();
+    });
+  });
+
+  describe('getModulesByType', () => {
+    test('should call to getModulesByType', async () => {
+      const expectedResult = ['string'];
+      const mockedParams = {
+        type: ModuleType.Burn,
+      };
+      // Mocked method
+      const mockedMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.getModulesByType).thenReturn(instance(mockedMethod));
+      // Stub the request
+      when(mockedMethod.callAsync(mockedParams.type)).thenResolve(expectedResult);
+
+      // Real call
+      const result = await target.getModulesByType(mockedParams);
+      // Result expectation
+      expect(result).toBe(expectedResult);
+      // Verifications
+      verify(mockedContract.getModulesByType).once();
+      verify(mockedMethod.callAsync(mockedParams.type)).once();
+    });
+  });
+
+  describe('getModule', () => {
+    test.todo('should fail as moduleAddress is not an Eth address');
+
+    test('should call to getModule', async () => {
+      const expectedString = 'Name';
+      const expectedNumbers = [new BigNumber(1), new BigNumber(2)];
+      const expectedResult = [
+        stringToBytes32(expectedString),
+        'stringstringstring',
+        'stringstringstring',
+        true,
+        expectedNumbers,
+      ];
+      const mockedParams = {
+        moduleAddress: '0x1111111111111111111111111111111111111111',
+      };
+      // Mocked method
+      const mockedMethod = mock(MockedCallMethod);
+      // Stub the method
+      when(mockedContract.getModule).thenReturn(instance(mockedMethod));
+      // Stub the request
+      when(mockedMethod.callAsync(mockedParams.moduleAddress)).thenResolve(expectedResult);
+
+      // Real call
+      const result = await target.getModule(mockedParams);
+      // Result expectation
+      expect(result.name).toBe(bytes32ToString(stringToBytes32(expectedString)));
```

Comment:
```suggestion
      expect(result.name).toBe(expectedString);
```

Reply:
[](https://imgur.com/NHcnDZD)

Reply:
`toEqual` maybe?

Reply:
This is weird code because something gets changed when converting twice, I recommended to Fede this

Reply:
toEqual doesn't work